File: selenium-java/src/test/java/frameworks/seleniumjava/pages/HomePage.java
package frameworks.seleniumjava.pages;

import org.openqa.selenium.WebDriver;
import io.qameta.allure.Step;

public final class HomePage extends BasePage {
  private static final String WELCOME = "welcome";
  private static final String WHOAMI = "whoami";

  public HomePage(WebDriver driver) {
    super(driver);
  }

  @Step
  public String getTitle() {
    return getTestIdText(WELCOME);
  }

  @Step
  public String getTextWhoAmI() {
    return getTestIdText(WHOAMI);
  }
}

File: selenium-python/pages/home_page.py
from __future__ import annotations

from pages.base_page import BasePage


class HomePage(BasePage):
    welcome = "welcome"
    whoami = "whoami"

    def get_title(self) -> str:
        return self.get_test_id_text(self.welcome)

    def get_text_who_am_i(self) -> str:
        return self.get_test_id_text(self.whoami)
